t/lib/CliRunners.pm

```diff
@@ -21,13 +25,28 @@ sub evalSucceeds {
     my ($jobset) = @_;
     my ($res, $stdout, $stderr) = captureStdoutStderr(60, ("hydra-eval-jobset", $jobset->project->name, $jobset->name));
     $jobset->discard_changes;  # refresh from DB
-    chomp $stdout; chomp $stderr;
-    print STDERR "Evaluation errors for jobset ".$jobset->project->name.":".$jobset->name.": \n".$jobset->errormsg."\n" if $jobset->errormsg;
-    print STDERR "STDOUT: $stdout\n" if $stdout ne "";
-    print STDERR "STDERR: $stderr\n" if $stderr ne "";
+    if ($res) {
+        chomp $stdout; chomp $stderr;
+        print STDERR "Evaluation errors for jobset ".$jobset->project->name.":".$jobset->name.": \n".$jobset->errormsg."\n" if $jobset->errormsg;
+        print STDERR "STDOUT: $stdout\n" if $stdout ne "";
+        print STDERR "STDERR: $stderr\n" if $stderr ne "";
+    }
     return !$res;
 }
 
+sub evalFails {
+    my ($jobset) = @_;
+    my ($res, $stdout, $stderr) = captureStdoutStderr(60, ("hydra-eval-jobset", $jobset->project->name, $jobset->name));
+    $jobset->discard_changes;  # refresh from DB
+    if (!$res) {
```

Comment:
Do we need this block? If so, I think the negation is wrong. If the eval fails, the exit code will (probably?) be non-zero, and the `if (!255) {` (`if (0) {`) block won't be entered. Thus, it'll only reach this when the eval succeeds, in which case `Evaluation errors for jobset` is incorrect, too.

Reply:
The goal is to print evaluation output if the evaluation succeeds. I think the message should be changed to say "Evaluation unexpectedly succeeded...." instead of "Evaluation errors". But otherwise I _think_ the code is right?

Reply:
Yeah, it was just the clash between those two that threw me off. "Evaluation unexpectedly succeeded" would be much clearer.